src/objects/module.rs

```diff
@@ -79,10 +79,24 @@ impl PyModule {
     /// Gets the module filename.
     ///
     /// May fail if the module does not have a `__file__` attribute.
+    #[allow(deprecated)]
```

Comment:
Detail: here I think it'd be better to allow the deprecation only for Python2. This can be done by duplicating the function body, I don't know of a better way:

```
    #[allow(deprecated)]
    pub fn filename<'a>(&'a self, py: Python) -> PyResult<&'a str> {
        unsafe { self.str_from_ptr(py, ffi::PyModule_GetFilename(self.0.as_ptr())) }
    }

    #[cfg(feature = "python3-sys")]
    pub fn filename<'a>(&'a self, py: Python) -> PyResult<&'a str> {
        unsafe { self.str_from_ptr(py, ffi::PyModule_GetFilename(self.0.as_ptr())) }
    }
```

I think it's tolerable in that case, because this is just a boilerplate wrapper. What do you think?

Reply:
Duplicating code seems a bit of a shame, but I guess it's probably the lesser of two evils.

Reply:
On second thought, the best would be to forward the deprecation to user code, not to have a suspicious warning in `rust-cpython` itself. This can be done in a follow-up, I'll try something.